pkg/kubelet/kubelet.go

```diff
@@ -101,8 +104,18 @@ func NewMainKubelet(
 	}
 
 	serviceStore := cache.NewStore(cache.MetaNamespaceKeyFunc)
-	if kubeClient != nil {
-		cache.NewReflector(cache.NewListWatchFromClient(kubeClient, "services", api.NamespaceAll, labels.Everything()), &api.Service{}, serviceStore).Run()
+	if kubeClient != nil && !reflect.ValueOf(kubeClient).IsNil() {
```

Comment:
> && !reflect.ValueOf(kubeClient).IsNil()

By changing kubeClient to be an interface, you broke an assumption that someone had. Please look through the places that set kubeClient and prevent them from storing a nil pointer in it. (This is the second landmine: https://gist.github.com/lavalamp/4bd23295a9f32706a48f)